exec/java-exec/src/main/java/org/apache/drill/exec/vector/complex/fn/JsonReaderUtils.java

```diff
@@ -120,4 +95,100 @@ public static void ensureAtLeastOneField(BaseWriter.ComplexWriter writer,
       }
     }
   }
+
+  /**
+   * Creates writers which correspond to the specified schema for specified root writer.
+   *
+   * @param writer      parent writer for writers to create
+   * @param columns     collection of columns for which writers should be created
+   * @param schema      table schema
+   * @param allTextMode whether all primitive writes should be for varchar type
+   */
+  public static void writeColumnsFromSchema(BaseWriter.ComplexWriter writer,
+      Collection<SchemaPath> columns, TupleMetadata schema, boolean allTextMode) {
+    Preconditions.checkArgument(schema != null, "Schema was not provided");
+    BaseWriter.MapWriter mapWriter = writer.rootAsMap();
+    for (SchemaPath column : columns) {
+      if (column.isDynamicStar()) {
+        writeColumnsForSchema(mapWriter, schema, allTextMode);
+      } else {
+        ColumnMetadata columnMetadata = schema.metadata(column.getRootSegmentPath());
+        writeToMap(mapWriter, column.getRootSegment(), columnMetadata, allTextMode);
+      }
+    }
+  }
+
+  private static void writeToMap(BaseWriter.MapWriter writer,
```

Comment:
Please reconsider naming and add java doc for all methods below.
Namings are confusing `writeToMap` vs `writeWithArray`, `writeColumnsFromSchema ` vs `writeColumnsForSchema`, it is not clear correlation between to / with, from / for.
Is better to have method that indicates what type of column is written rather than overloaded `writeColumn`, this way user can see from code what column will be written.

Reply:
Tried to think up better names, but at least removed these confusing prepositions from some method names. Also, got rid of overloaded `writeColumn`.